task_master.go

```diff
@@ -628,7 +640,9 @@ func (tm *TaskMaster) forkPoint(p models.Point) {
 }
 
 func (tm *TaskMaster) WritePoints(database, retentionPolicy string, consistencyLevel imodels.ConsistencyLevel, points []imodels.Point) error {
-	if tm.closed {
+	tm.writesMu.RLock()
+	defer tm.writesMu.RUnlock()
+	if tm.writesClosed {
```

Comment:
To safely close the writePointsIn edge we need to ensure that no more writes were happening.  This meant that we needed to check a RWMutex and perform a multistage close.

The close process looks something like this:

1. Stop all writes
2. Close writePointsIn edge.
3. Wait for all forks to finish
4. Delete forks
5. Stop tasks
6. Mark task master as closed.

This should  have been the process all along but previous with just the HTTP method of writing points it didn't matter. But now with the new loopback way of writing points this explicit close process was needed to prevent panics on writing to the closed edge.

In other words the HTTPD service was always closed before the TaskMaster so the fact that the Close operation wasn't safe didn't matter since writes could not come in during the close operation. But this new loopback method for writes changed that.